vms/platformvm/state/merkle_state.go

```diff
@@ -0,0 +1,1706 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/cache/metercacher"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/prefixdb"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	"github.com/ava-labs/avalanchego/vms/platformvm/config"
+	"github.com/ava-labs/avalanchego/vms/platformvm/metrics"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/platformvm/status"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+)
+
+const (
+	HistoryLength = int(256)
+
+	valueNodeCacheSize        = 512 * units.MiB
+	intermediateNodeCacheSize = 512 * units.MiB
+	utxoCacheSize             = 8192 // from avax/utxo_state.go
+)
+
+var (
+	_ State = (*merkleState)(nil)
+
+	merkleStatePrefix      = []byte{0x00}
+	merkleSingletonPrefix  = []byte{0x01}
+	merkleBlockPrefix      = []byte{0x02}
+	merkleBlockIDsPrefix   = []byte{0x03}
+	merkleTxPrefix         = []byte{0x04}
+	merkleIndexUTXOsPrefix = []byte{0x05} // to serve UTXOIDs(addr)
+	merkleUptimesPrefix    = []byte{0x06} // locally measured uptimes
+	merkleWeightDiffPrefix = []byte{0x07} // non-merklelized validators weight diff. TODO: should we merklelize them?
+	merkleBlsKeyDiffPrefix = []byte{0x08}
+
+	// merkle db sections
+	metadataSectionPrefix      = byte(0x00)
+	merkleChainTimeKey         = []byte{metadataSectionPrefix, 0x00}
```

Comment:
What's the advantage of manually prefixing this instead of using a prefixdb for these? Just because it's faster?